src/fpe/fpe.ts

```diff
@@ -0,0 +1,163 @@
+  alphabet?: string
+  additionalCharacters?: string
+  radix?: number
+  digits?: number
+}
+
+// Use an enum to define the valid alphabet IDs
+enum FpeAlphabetId {
+  Numeric = "numeric",
+  Hexadecimal = "hexa_decimal",
+  AlphaLower = "alpha_lower",
+  AlphaUpper = "alpha_upper",
+  Alpha = "alpha",
+  AlphaNumeric = "alpha_numeric",
+  Utf = "utf",
+  Chinese = "chinese",
+  Latin1Sup = "latin1sup",
+  Latin1SupAlphanum = "latin1sup_alphanum",
+}
+
+/**
+ * This Format Preserving Encryption (FPE) function provides FPE-techniques for use in a zero-trust environment. These techniques are based on FPE-FF1 which is described in [NIST:800-38G](https://nvlpubs.nist.gov/nistpubs/specialpublications/nist.sp.800-38g.pdf).
+ */
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export async function Fpe() {
+  if (initialized === undefined) {
+    if (wasmInit === undefined) {
+      throw new Error("Please provide a WASM init function")
+    }
+
+    const loadModule = wasmInit()
+    initialized = init(loadModule).then(() => undefined)
+  }
+
+  await initialized
+
+  // This encryption function takes plaintext as number, bigint or string:
+  //
+  // * when encrypting an integer (bigint typescript datatype) and big integer (represented as string), the radix and digits are needed to be specified. The `max_value` is calculated as the number of `digits` raised to the power of `radix`.
+  // * when encrypting a float, no option is required
+  // * when encrypting a string, an alphabet id can be given to specify the alphabet required:
+  //   - "numeric": 0123456789
+  //   - "hexa_decimal": 0123456789abcdef
+  //   - "alpha_lower": abcdefghijklmnopqrstuvwxyz
+  //   - "alpha_upper": ABCDEFGHIJKLMNOPQRSTUVWXYZ
+  //   - "alpha": abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ
+  //   - "alpha_numeric": 0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ
+  //   - "utf": creates an Alphabet with the first 63489 (~2^16) Unicode characters
+  //   - "chinese": creates an Alphabet with the Chinese characters
+  //   - "latin1sup": creates an Alphabet with the latin-1 and latin1-supplement characters (supports French)
+  //   - "latin1sup_alphanum": creates an Alphabet with the latin-1 and latin1-supplement characters but without the non alphanumeric characters (supports French)
+  const encrypt = async (
+    key: Uint8Array,
+    tweak: Uint8Array,
+    plaintext: string | number | bigint,
+    options: FpeOptions = {},
+  ): Promise<string | number | bigint> => {
+    if (typeof plaintext === "string") {
+      // There are 2 cases where we have string a FPE-plaintext
+      // - string as big integer where specifying the radix and digits is required
+      // - other strings where specifying an alphabet may be required
+      if (
+        typeof options.radix !== "undefined" &&
+        typeof options.digits !== "undefined"
+      ) {
+        return webassembly_fpe_encrypt_big_integer(
+          plaintext,
```

Comment:
alphabet is not used here ?

Reply:
no alphabet is used for big int encryption.
      // There are 2 cases where we have string a FPE-plaintext
      // - string as big integer where specifying the radix and digits is required
      // - other strings where specifying an alphabet may be required
if no radix nor digits is specified, the string input is encrypted as a "real" string

Reply:
is it any clearer?

Reply:
yes - but when will this case happen : typeof plaintext === "bigint" ? bigints can be string or bigints at first?

Reply:
this case happens when user explicitly gave a BigInt type (ex: plaintext = BigInt(123456)). Otherwise it is a `number` type.

For big integers (with digits number > 10), they should be given in string representation.